NMS-13364: Table formatting issues in new docs (#3531)
OpenNMS/opennms: docs/modules/operation/pages/service-assurance/monitors/DiskUsageMonitor.adoc

= DiskUsageMonitor

> The DiskUsageMonitor tests the amount of free space available on certain node storage.

Reviewer: ```suggestion
The DiskUsageMonitor tests the amount of free space available on the specified storage disk of a monitored node.
```

The monitor inspects the http://tools.ietf.org/html/rfc1514[HOST-RESOURCES-MIB] hrStorageTable to get information about the available free storage space.

> A storage's description (as found in the corresponding hrStorageDescr object) must match the criteria the `disk` and `match-type` parameters specify to be monitored.

Reviewer: ```suggestion
A disk's description (as found in the corresponding hrStorageDescr object) must match the criteria the `disk` and `match-type` parameters specify to be monitored.
```

> A storage's available free space is calculated using the corresponding hrStorageSize and hrStorageUsed objects.

Reviewer: ```suggestion
The available free space is calculated using the corresponding hrStorageSize and hrStorageUsed objects.
```

CAUTION: The hrStorageUsed does not account for filesystem reserved blocks (for example, for the super-user), so DiskUsageMonitor will report the service as 
         unavailable only when the amount of free disk space is actually lower than `free` minus the percentage of reserved filesystem blocks.

This monitor uses SNMP to accomplish its work.
Therefore, systems against which you use it must have an SNMP agent supporting the HOST-RESOURCES-MIB installed and configured.
Most modern SNMP agents, including most distributions of the _Net-SNMP agent_ and the _SNMP service_ that ships with Microsoft Windows, support this MIB.
Out-of-the-box support for HOST-RESOURCES-MIB among commercial Unix operating systems may be spotty.

== Monitor facts

[options="autowidth"]
|===
| Class Name     | `org.opennms.netmgt.poller.monitors.DiskUsageMonitor`

> | Remote Enabled | false, relies on SNMP configuration

Reviewer: Is this true?  Shouldn't this work remotely via Minions?

Author: @indigo423, do you know the answer to this?

Author: Chandra confirmed it should be "true".

== DiskUsageMonitor vs. thresholds

> You can also use thresholds to monitor a storage's available free space if you are already collecting these data.

Reviewer: ```suggestion
You can also use thresholds to monitor a storage's available free space if you are already collecting these data.
The difference between thresholds and this monitor is that while both can trigger notifications, using this monitor will put the node into a service down state and trigger an outage.
```
